### PR TITLE
BOAC-2904: Displays appointments in search results

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2904

### CODE REVIEW

File: src/components/search/AdvisingNoteSnippet.vue [hunk 1]
```diff
@@ -48,26 +48,3 @@ export default {
...
-.advising-note-search-result-footer {
-  color: #999;
-  font-size: 14px;
-}
-.advising-note-search-result-snippet {
-  font-size: 16px;
-  line-height: 1.3em;
-  margin: 5px 0;
-}
-</style>
```
Comment (johncrossman) [inline src/components/search/AdvisingNoteSnippet.vue]: @lyttam  Are these styles no longer used?

Comment (johncrossman) [reply src/components/search/AdvisingNoteSnippet.vue]: Nevermind.  I see them below.

File: src/components/search/AppointmentSnippet.vue [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<template>
+  <div
+    :id="`advising-note-search-result-${appointment.id}`"
```
Comment (johncrossman) [inline src/components/search/AppointmentSnippet.vue]: @lyttam  These id prefixes should be `appointment-search-result-...`, yes?

File: src/components/search/AppointmentSnippet.vue [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    class="advising-note-search-result"
+    :class="{'demo-mode-blur': user.inDemoMode}">
+    <h3 class="advising-note-search-result-header">
+      <router-link
+        :id="`link-to-student-${appointment.student.uid}`"
```
Comment (johncrossman) [inline src/components/search/AppointmentSnippet.vue]: @lyttam  The id above might collide with other search result items.

File: src/components/search/AppointmentSnippet.vue [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  props: {
+    appointment: Object,
+  },
+  data: () => ({
+    lastModified: undefined
+  }),
+  created() {
+    const timestamp = this.get(this.appointment, 'updatedAt') || this.get(this.appointment, 'createdAt');
+    if (timestamp) {
+      this.lastModified = this.$moment(timestamp).tz(this.timezone);
```
Comment (johncrossman) [inline src/components/search/AppointmentSnippet.vue]: @lyttam and I talked to Flint and decided to toss out `updatedAt` above. For now, `createdAt` will be the date of the appt.

File: src/views/Search.vue [hunk 5]
```diff
@@ -170,6 +208,24 @@ export default {
...
+        false,
+        false,
+        this.extend({}, this.noteAndAppointmentOptions, this.appointmentOptions),
+        null
+      )
+        .then(data => {
+          this.results.appointments = this.concat(this.results.appointments, data.appointments);
+          this.loadingAdditionalAppointments = false;
+        });
+    },
```
Comment (johncrossman) [inline src/views/Search.vue]: @lyttam - A screen-reader alert, after results load, would be good. (Maybe `fetchMoreNotes()` needs the same treatment.)

File: boac/models/appointment.py [hunk 6]
```diff
@@ -308,9 +318,8 @@ def _to_json(search_result):
         'createdBy': search_result['created_by'],
         'dept_code': search_result['dept_code'],
```
Comment (johncrossman) [inline boac/models/appointment.py]: @lyttam - A pre-existing problem: camelCase needed in key above.

File: src/components/search/AppointmentSnippet.vue (hunk 2 above)
Comment (lyttam) [reply src/components/search/AppointmentSnippet.vue]: @johncrossman Yes, will amend.

File: src/components/search/AppointmentSnippet.vue (hunk 3 above)
Comment (lyttam) [reply src/components/search/AppointmentSnippet.vue]: @johncrossman Good catch, I will make sure those IDs are unique

**Comment by johncrossman** [comment]
Thank you @lyttam  I posted a few comments. Let me know when amendments are in.

File: src/views/Search.vue (hunk 5 above)
Comment (lyttam) [reply src/views/Search.vue]: I tested this with a screen reader, and the `SectionSpinner` mixin takes care of the screen reader alert for both notes and appointments.

**Comment by lyttam** [comment]
Ready for another look, @johncrossman.

**Comment by johncrossman** [comment]
Thanks @lyttam

**Pull request merged**